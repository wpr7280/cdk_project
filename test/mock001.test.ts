import {App} from "aws-cdk-lib";
import {Template} from "aws-cdk-lib/assertions";
import {Code, CodeConfig} from "aws-cdk-lib/aws-lambda";
// import {MyLambdaStack} from "../bin/cdk-project";

describe('Mock Asset Test', () => {
    // let fromAssetMock: jest.SpyInstance ;
    // beforeAll(() => {
    // mock the Code calls so tests run quicker
    // let fromAssetMock = jest.spyOn(Code, 'fromAsset').mockReturnValue({
    //     isInline: false,
    //     bind: (): CodeConfig => {
    //         return {
    //             s3Location: {
    //                 bucketName: 'my-bucket',
    //                 objectKey: 'my-key',
    //             },
    //         };
    //     },
    //     bindToResource: () => {
    //         return;
    //     },
    // } as any);
    // // });
    // // afterAll(() => {
    // //     // restore the Code from mock
    // //     fromAssetMock?.mockRestore();
    // // });
    // const app = new App();
    // const lambda = new MyLambdaStack(app, 'mockLambdaStack', {});
    // const template = Template.fromStack(lambda);
    // test('check snapshot', () => {
    //     expect(template).toMatchSnapshot();
    // });
})
