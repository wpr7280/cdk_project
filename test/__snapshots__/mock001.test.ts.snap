// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mock Asset Test check snapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "MyAssetsFunctionA132C47B": Object {
      "DependsOn": Array [
        "MyAssetsFunctionServiceRoleE4026214",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": "my-bucket",
          "S3Key": "my-key",
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "MyAssetsFunctionServiceRoleE4026214",
            "Arn",
          ],
        },
        "Runtime": "nodejs16.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "MyAssetsFunctionServiceRoleE4026214": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
